added DMA write to SD card drivers

mbeckersys opened the pull request:
Successfully tested on a STM32F427VIT and merged back into this library for public use.

The TX_Active bit is hanging around for a bit, leaving a single DMA transfer to be a bit slow, yet (~2msec).

lambourg commented:
Thank you very much for this contribution :)

Fabien-Chouteau commented:
Thanks Martin!

Fabien-Chouteau commented:
Martin, this patch broke the STM32F7 builds:

`stm32-sdmmc.adb:1507:17: "Startbit_Not_Detected" is undefined (more references follow)
stm32-sdmmc.adb:1550:17: "DMA_Alignment_Error" is undefined (more references follow)
stm32-sdmmc.adb:1623:04: (style) subprogram body has no previous spec
`

STM32F7X and STM32F7x9 use the `drivers/sd/sdio` implementation.

mbeckersys commented:
Sorry for that. I did not realize that is has a different specification.
I added the missing definitions in ARM/STM32/drivers/sd/sdmmc/stm32-sdmmc.ads as well,  which should fix it. 

However, there was another problem with the STM32F7 build, which was not related to my pull request:
`
device_sd_configuration.ads:32:39: "SDMMC1_Peripheral" not declared in "SDMMC"
device_sd_configuration.ads:32:40: possible misspelling of "SDMMC_Peripheral"
device_sd_configuration.ads:33:41: "SDMMC1_Periph" not declared in "SDMMC"
device_sd_configuration.ads:33:42: possible misspelling of "SDMMC_Periph"
`

I fixed both and did another pull request #34 with those changes.
